### PR TITLE
You can no longer reload the TAT by yourself

### PR DESCRIPTION
**Pull request by Three333Digit**
## About The Pull Request
Title.

[Video demonstration](https://streamable.com/ptx14e)

## Why It's Good For The Game
According to this #9650 

>Adds the TAT-36 AT Gun, a delpoyed and moveable weapon that is meant to be used by mutliple people to be used effectively.
>meant to be used by mutliple people to be used effectively.

## Changelog
:cl:
bal: You can no longer reload the TAT by yourself
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/mounted.dm [hunk 1]
```diff
@@ -457,6 +457,13 @@
 	. = ..()
 	sponson = new sponson(src)
 
+/obj/machinery/deployable/mounted/moveable/atgun/attackby(obj/item/I, mob/user, params)
+	if(user.interactee)
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/mounted.dm:461]: this is just checking if the user is interacting with anything, not src specifically.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by Pariah919** [comment]
You still need two people to operate this thing to maximum efficency anyway. Loading with someone else is always going to be faster, this is a mostly unnecessary in my opinion

File: code/modules/projectiles/guns/mounted.dm (hunk 1 above)
Comment (Three333Digit) [reply code/modules/projectiles/guns/mounted.dm]: Is this okay?

**Comment by Lumipharon** [comment]
Also this should probs just be a balloon alert.

**Comment by TChalkman** [comment]
Unnecessary change. This will turn the AT-36 useless. 
TAT is already a weapon barely anybody uses because of its issues.

I could start writing a very long list of issues and annoying shit, which is the reason it's used by literally 2 people. But this isn't the topic.

This change will basically double the delay time between shots unless you get help, will make the weapon TWICE as annoying to work with, and it's going to need yet another babysitter when AT users already need a general help by marines and constant supply from RO.
Also, who the fuck finds fun to see a dude shoot and only load ammo? Who is gonna want to help you besides in some random cases when everybody is stuck in a choke or fob?

If this gets merged, we are going to go from 1-2 TAT players to extinct.

**Comment by Three333Digit** [comment]
> Also this should probs just be a balloon alert.

Done. Why the fuck is it failing integration tests though?

**Comment by blackdav123** [comment]
you can literally rapid fire TAT shells with just the loading time as cooldown

these cost nearly nothing and are the best weapon marines have for securing xeno kills

sorry you'll have to work with somebody else for 100% efficiency.

**Comment by Losenis** [comment]
Salt pr due to not thinking at the actual consequences of this. It'll only make people just skip the scope and zoom mechanics and use it almost pb to avoid the jank of having your client resize twice for every single shot you take.

**Comment by blackdav123** [comment]
2 second delay between massive AOE damage that cant cause any friendly fire is broken

no other weapon in the game is capable of dealing this much damage this quickly with zero risk of friendly fire

forcing this fire support weapon to require two players to use effectively just like mortar and howie does makes a lot of sense.

**Comment by Losenis** [comment]
>massive AoE damage
>that cant cause any friendly fire

Bro are you high, I lost count how many times someone just killed a guy by shooting the roony on top of him with a fucking HE. APCR is popular for a reason.

**Comment by maikilangiolo** [comment]
> massive AOE damage that cant cause any friendly fire

Do you think people would do that? Go on the internet and lie?

**Comment by Passe2142** [comment]
Mortar/Howie/MRLS with a buddy works better, TAT didn't need it. It should work better with a buddy. Not a nerf per se, more like the intended idea.

File: code/modules/projectiles/guns/mounted.dm [hunk 2]
```diff
@@ -457,6 +457,13 @@
 	. = ..()
 	sponson = new sponson(src)
 
+/obj/machinery/deployable/mounted/moveable/atgun/attackby(obj/item/I, mob/user, params)
+	if(user.interactee == src)
+		balloon_alert(user, "Cannot reload while you're manning it")
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/mounted.dm:462]: Oh this is also not checking for what I is. So the balloon alert doesn't really make sense if you're smacking it with some other item.

**Comment by Al-1ce** [comment]
As a concept I love this but I know in reality it means the TAT will never see the light of day again.

**Review by Lumipharon — APPROVED** [review]

**Comment by FosterGrime-Heymynameslouie** [comment]
already a bit too late but cant you just shoot, get out of tat, reload it yourself, then man it again?

**Comment by SpanishBnn** [comment]
> already a bit too late but cant you just shoot, get out of tat, reload it yourself, then man it again?

Yeah, but it would also mean an alone TAT player will take double/triple the time shoot again
Shoot>get off>load round>get back in>aim and zoom>repeat
Instead of 
Shoot>reload>repeat

**Comment by SpanishBnn** [comment]
And nice, awful PR got merged, guess I’m back at only playing rocket sledge 
Wonder who’s gonna be using the TAT consistently now, if at all

**Comment by Losenis** [comment]
It only makes the solo TAT skip aim and zoom and instead drag the AT closer and call skill issue on whoever's dumb enough to stand in front of the TAT. It doesn't achieve what it attempts to achieve and it should've been obvious from the start.

**Comment by chizzy376** [comment]
salt pr

**Pull request merged**